plugins/modules/swim_workflow_manager.py

```diff
@@ -1513,78 +1532,121 @@ def get_diff_tagging(self):
             This function tags or untags a software image as a golden image in Cisco Catalyst Center based on the provided
             tagging details. The tagging action is determined by the value of the 'tagging' attribute
-            in the 'tagging_details' dictionary.If 'tagging' is True, the image is tagged as golden, and if 'tagging'
+            in the 'tagging_details' dictionary. If 'tagging' is True, the image is tagged as golden, and if 'tagging'
             is False, the golden tag is removed. The function sends the appropriate request to Cisco Catalyst Center and updates the
             task details in the 'result' dictionary. If the operation is successful, 'changed' is set to True.
         """
 
         tagging_details = self.want.get("tagging_details")
         tag_image_golden = tagging_details.get("tagging")
         image_name = self.get_image_name_from_id(self.have.get("tagging_image_id"))
+        device_role = tagging_details.get("device_role", "ALL")
+        device_role_no, already_un_tagged_device_role, already_tagged_device_role = [], [], []
 
-        image_params = dict(
-            image_id=self.have.get("tagging_image_id"),
-            site_id=self.have.get("site_id"),
-            device_family_identifier=self.have.get("device_family_identifier"),
-            device_role=tagging_details.get("device_role", "ALL").upper()
-        )
+        device_roles = ["core", "distribution", "access", "border router", "unknown", "all"]
 
-        response = self.dnac._exec(
-            family="software_image_management_swim",
-            function='get_golden_tag_status_of_an_image',
-            op_modifies=True,
-            params=image_params
-        )
-        self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
+        for role in device_role.split(','):
+            role = role.strip()
+            device_role_no.append(role)
 
-        response = response.get('response')
-        if response:
-            image_status = response['taggedGolden']
-            if image_status and image_status == tag_image_golden:
+            if role.lower() not in device_roles:
+                self.status = "failed"
+                self.msg = (
+                    "Validation Error: The specified device role '{0}' is not recognized. "
+                    "Please ensure the role matches one of the known device roles: {1}."
+                ).format(role, ", ".join(device_roles))
+                self.log(self.msg, "ERROR")
+                self.result['response'] = self.msg
+                self.check_return_status()
+
+        for role in device_role.split(','):
+            image_params = {
+                "image_id": self.have.get("tagging_image_id"),
+                "site_id": self.have.get("site_id"),
+                "device_family_identifier": self.have.get("device_family_identifier"),
+                "device_role": role.upper()
+            }
+
```

Comment:
self.log("Parameters for checking tag status for role '{0}': {1}".format(role, image_params), "DEBUG")

Reply:
updated the code